ui/src/design/RapiDoc/index.tsx

```diff
@@ -0,0 +1,37 @@
+import "./styles/RapiDoc.css";
+import "rapidoc";
+
+import { useEffect } from "react";
+
+interface RapiDocProps {
+  spec: string | object;
+  className?: string;
+}
+
+export function RapiDoc({ spec, className }: RapiDocProps) {
+  useEffect(() => {
```

Comment:
I would love to remove this `useEffect` here. I'm not a fan of `useEffect` and I try to avoid it as much as I can. Technically, it is correct to use it here for the `loadSpec` part as we need to access the DOM node which is not a managed by React component.

Still, there is a slightly more elegant way to solve this using a [ref](https://tkdodo.eu/blog/avoiding-use-effect-with-callback-refs) (my assumption is that we remove the `spec-url` part here and only allow to pass an object) 

```typescript
 <rapi-doc
      ref={(rapiDocElement) => rapiDocElement?.loadSpec(spec)}  // 👈 this is new
      id="rapidoc"
      render-style="read"
      allow-try="false"
      allow-authentication="false"
      show-header="false"
      show-info="true"
      theme="light"
      primary-color="#5364FF"
    />
``` 

To make typescript happy, we need to add this to the types.d.ts.

```typescript
declare namespace JSX {
  interface IntrinsicElements {
    "rapi-doc": {
      id?: string;
      theme?: string;
      ref?: React.Ref<RapiDocElement>; // 👈 this is new
    };
  }
}

```

Reply:
More elegant indeed. 🤩